public/app/plugins/panel/geomap/GeomapPanel.tsx

```diff
@@ -267,6 +267,8 @@ export class GeomapPanel extends Component<Props, State> {
         this.applyLayerFilter(state.handler, state.options);
       }
     }
+    // Because data changed, check map view and change if needed (data fit)
+    this.map!.setView(this.initMapView(this.props.options.view, this.map!.getLayers()));
```

Comment:
Should add conditional for data fit, so this view init only gets called for that view setting.